android/src/com/mapswithme/maps/MwmApplication.java

```diff
@@ -161,7 +160,30 @@ public String getApkPath()
     }
   }
 
-  public String getDataStoragePath()
+  private static boolean isMapStorage(String path)
+  {
+    return !TextUtils.isEmpty(path) && Utils.directoryExists(path) &&
+           new File(path, "World.mwm").exists();
```

Comment:
World.mwm not necessarily exists there. It can be also in the APK, or in OBB file. Why do you check for it?